diffrax/brownian/tree.py

```diff
@@ -88,70 +152,224 @@ def __init__(
+
+        # Rescaling back from [0, 1] to the original interval [t0, t1].
+        interval_len = self.t1 - self.t0  # can be any dtype
+        sqrt_len = jnp.sqrt(interval_len)
+
+        def sqrt_mult(z):
+            # need to cast to dtype of each leaf in PyTree
+            dtype = jnp.dtype(z)
+            return (z * sqrt_len).astype(dtype)
+
+        def mult(z):
+            dtype = jnp.dtype(z)
+            return (interval_len * z).astype(dtype)
+
+        return LevyVal(
+            dt=jtu.tree_map(mult, x.dt),
+            W=jtu.tree_map(sqrt_mult, x.W),
+            H=jtu.tree_map(sqrt_mult, x.H),
+            bar_H=None,
+            K=jtu.tree_map(sqrt_mult, x.K),
+            bar_K=None,
+        )
+
     @eqx.filter_jit
     def evaluate(
-        self, t0: Scalar, t1: Optional[Scalar] = None, left: bool = True
-    ) -> PyTree[Array]:
-        del left
+        self,
+        t0: Scalar,
+        t1: Optional[Scalar] = None,
+        left: bool = True,
+        use_levy: bool = False,
+    ) -> LevyVal:
+        def _is_levy_val(obj):
+            return isinstance(obj, LevyVal)
+
         t0 = eqxi.nondifferentiable(t0, name="t0")
+        # map the interval [self.t0, self.t1] onto [0,1]
+        t0 = linear_rescale(self.t0, t0, self.t1)
+        levy_0 = self._evaluate(t0)
         if t1 is None:
-            return self._evaluate(t0)
+            levy_out = levy_0
+
         else:
             t1 = eqxi.nondifferentiable(t1, name="t1")
-            return jtu.tree_map(
-                lambda x, y: x - y,
-                self._evaluate(t1),
-                self._evaluate(t0),
-            )
+            # map the interval [self.t0, self.t1] onto [0,1]
+            t1 = linear_rescale(self.t0, t1, self.t1)
+            levy_1 = self._evaluate(t1)
+
+            levy_out = jtu.tree_map(_levy_diff, levy_0, levy_1, is_leaf=_is_levy_val)
 
-    def _evaluate(self, τ: Scalar) -> PyTree[Array]:
-        map_func = lambda key, shape: self._evaluate_leaf(key, τ, shape)
+        levy_out = levy_tree_transpose(self.shape, self.levy_area, levy_out)
+        # now map [0,1] back onto [self.t0, self.t1]
+        levy_out = self._denormalise_bm_inc(levy_out)
+        assert isinstance(levy_out, LevyVal)
+        return levy_out if use_levy else levy_out.W
+
+    def _evaluate(self, r: Scalar) -> PyTree[LevyVal]:
+        """Maps the _evaluate_leaf function at time τ using self.key onto self.shape"""
+        r = eqxi.error_if(
+            r,
+            r < 0,
+            "Cannot evaluate VirtualBrownianTree outside of its range [t0, t1].",
+        )
+        r = eqxi.error_if(
+            r,
+            r > 1,
+            "Cannot evaluate VirtualBrownianTree outside of its range [t0, t1].",
+        )
+        # Clip because otherwise the while loop below won't terminate, and the above
+        # errors are only raised after everything has finished executing.
+        r = jnp.clip(r, 0, 1)
```

Comment:
I know you were copying this from what's already here, but actually this is no longer true! Let's fix this up whilst we're here. We can remove this line.
Likewise, we can actually improve efficiency a little bit by having just a single `error_if` combining both conditions:
```python
  r = eqxi.error_if(
      r,
      (r < 0) | (r > 1),
      "Cannot evaluate VirtualBrownianTree outside of its range [t0, t1].",
  )
```

Reply:
I don't really understand why this was an issue before, and is not now. Even before this, the distance between s and u would have eventually become smaller than `tol`, no? Anyway, yes I made the change :)